### PR TITLE
Update account details page

### PR DESCRIPTION
**Pull request by Fabtron**
### What's new?

The account details page has been changed accoringly to the requirements in APPS-1277 and a dialog has been added to change the the card name.

### How to test?

Launch the app, add a card and checkout the details page

### CODE REVIEW

**Comment by cmaier** [comment]
Beim mehrmaligen Bearbeiten des Titels hinternander steht immer der erste Titel da.

Zusätzlich wäre es schön, wenn der Titel Komplett selektiert ist. Das erleichtert das vollständige Renaming. Was meinst du dazu?

File: app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                    Text(
+                        modifier = Modifier
+                            .padding(start = 8.dp),
+                        text = stringResource(id = R.string.change_title_dialog_button),
+                        style = MaterialTheme.typography.labelLarge
+                    )
+                }
+            }
+        }
+    }
```
Comment (cmaier) [inline app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt:82]: Maybe we can merge this Dialog and the Dialog from [here](https://github.com/snabble/snabble-pay-android-sdk/blob/82b2b93c92a6f426dc7edcd3e3d68c15ad29fcd4/app/src/main/java/io/snabble/pay/app/ui/SnabblePayDialog.kt#L44) to one implementation 👍

**Review by cmaier — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt (hunk 1 above)
Comment (Fabtron) [reply app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt:82]: I don't think we should merge everything simply bevause we can. The dialogs implement completly different usescase and thus shouldn't be merged to one.

**Comment by Fabtron** [comment]
> Beim mehrmaligen Bearbeiten des Titels hinternander steht immer der erste Titel da.
> 
> Zusätzlich wäre es schön, wenn der Titel Komplett selektiert ist. Das erleichtert das vollständige Renaming. Was meinst du dazu?

I don't see the requirement here as the title states edit label not change lable. I also don't know any cases where this is applied.
I will look into the other problem.

**Comment by Fabtron** [comment]
> Beim mehrmaligen Bearbeiten des Titels hinternander steht immer der erste Titel da.
> 
> Zusätzlich wäre es schön, wenn der Titel Komplett selektiert ist. Das erleichtert das vollständige Renaming. Was meinst du dazu?

The other problem is fixed.

**Comment by cmaier** [comment]
> I don't see the requirement here as the title states edit label not change lable. I also don't know any cases where this is applied.
There is no requirement for it, but eventually using the feature made me making this suggestion. If you think it's inappropriate and conflicts with the requirements, I'm fine leaving it as it is. :)

File: app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt (hunk 1 above)
Comment (cmaier) [reply app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt]: To be fair, it has nothing todo with we can and cannot do. It's about consistency throughout the app.

See, both dialogs as they have been implemented according to "requirements":
|#35|#37 _(this PR)_|
|--|--|
|Screenshot 2024-01-22 at 09 05 21|Screenshot 2024-01-22 at 09 07 12|

I don't think that this is what we really want. I'd say latest changes are a step towards the right direction about appearance of the dialogs:

|#35|#37 _(this PR)_|
|--|--|
|Screenshot 2024-01-22 at 09 12 45|Screenshot 2024-01-22 at 09 07 12|

What do you think?

Comment (Fabtron) [reply app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt]: I simply say we should not merge the implementations together. We have a simply alert dialog with the functionality to accept or decline and a dialog to edit something. Completlly differen purposes wich has nothing to do with the otherall desing. 
If u refere to my old implemenation of the alert dialog and the design difference to the edit dialog, i agree with you. But these changes have been applied in the other PR. If u compare the last given images the design matches. Also this dialog has been approved as u can read in the comment section of the affected ticket.

File: app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/AccountDetails.kt [hunk 2]
```diff
@@ -46,77 +51,113 @@ fun AccountDetails(
...
+            }
+        ) {
+            if (showChangeTitleDialog.value) {
+                SnabblePayDialog(
+                    dialogTitle = stringResource(id = R.string.account_details_change_title),
+                    secondaryButtonLabel = stringResource(id = R.string.change_title_dialog_button),
+                    onDismiss = {
+                        onLabelChange(cardName.value, accountCard.cardBackgroundColor)
+                        showChangeTitleDialog.value = false
+                    }
```
Comment (cmaier) [inline app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/AccountDetails.kt:76]: The change dialog now has the problem, that the if the secondary button is used for the primary action, a dismiss of the dialog is also like confirming. That is uncommon and should not be the case. I guess the intention was to use the style from the secondary button as primary action, like in the design, but I'd recommend sorting this out with UI/UX that the primary action button should appear in the primary action style.

Comment (Fabtron) [reply app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/AccountDetails.kt:76]: changed

Comment (cmaier) [reply app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/AccountDetails.kt:76]: There is no change yet!

Comment (Fabtron) [reply app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/AccountDetails.kt:76]: Oh i forgot to push...

**Comment by cmaier** [comment]
If a tap to change the title, do some changes, but dismiss the dialog, the changes still appear on the details screen. On clicking change title again, the unsaved changes appear in the dialog.

**Comment by Fabtron** [comment]
> If a tap to change the title, do some changes, but dismiss the dialog, the changes still appear on the details screen. On clicking change title again, the unsaved changes appear in the dialog.

Oh uses the same state for the card name.  Fixed it!

**Review by cmaier — APPROVED** [review]
Nice, works!

**Pull request merged**